src/entities/createPool/types.ts

```diff
@@ -59,6 +59,14 @@ export type CreatePoolV3StableInput = CreatePoolV3BaseInput & {
         paysYieldFees: boolean;
     }[];
 };
+
+export type CreatePoolStableSurgeInput = Omit<
+    CreatePoolV3StableInput,
+    'poolHooksContract' | 'poolType'
```

Comment:
Omiting the poolHooksContract since there is a canonical BLabs one, I assume?

Reply:
Looks like the new `StableSurgeFactory` couples the factory's deployment with the hook :eyes:

https://github.com/balancer/balancer-v3-monorepo/blob/4680d59ede505847163197885780169af8873ec1/pkg/pool-hooks/contracts/StableSurgePoolFactory.sol#L27-L39

Reply:
Perhaps we might consider changing `poolHooksContract` to be an optional property on the `CreatePoolV3BaseInput` type and then doing like 

```
input.poolHooksContract ?? zeroAddress,
```

for the createPoolStable and Weighted?

https://github.com/balancer/b-sdk/blob/4f07d199b78720816c63743efd6afae67aa7f21f/src/entities/createPool/createPoolV3/stable/createPoolStable.ts#L45

Reply:
With the goal of making it easier for the user or something else?

Reply:
yeah would abstract away the need for user to know that they have to pass the zero address if they don't want to use a hook when creating pools with the core stable and weighted factories

Also to prevent the need for adding `<Omit ..., 'poolHooksContract` each time a new pool factory tied to a hook is added